### PR TITLE
netlist: Fix standalone nltool building with nvcc.

### PR DESCRIPTION
**Pull request by couriersud**
nvcc 11.3 (latest cuda tools) has an issue with some auto x(some
variable) declariations. This commit fixes the compile errors.

### CODE REVIEW

File: src/lib/netlist/plib/pstream.h [hunk 1]
```diff
@@ -299,7 +299,7 @@ class pbinary_writer
 	template <typename T>
 	void write(const std::vector<T> &val)
 	{
-		const auto sz(val.size());
+		const T sz(val.size());
```
Comment (cuavas) [inline src/lib/netlist/plib/pstream.h:302]: This shouldn’t be `T` – it’s `std::vector<T>::size_type` (in practice it’s `std::size_t` unless `std::vector` is specialised for `T`).

File: src/lib/netlist/plib/pstring.cpp [hunk 2]
```diff
@@ -83,7 +83,7 @@ typename pstring_t<F>::size_type pstring_t<F>::find(const pstring_t &search, siz
 	auto istart = std::next(begin(), static_cast<difference_type>(start));
 	for (; istart != end(); ++istart)
 	{
-		auto itc(istart);
+		auto itc =istart;
```
Comment (cuavas) [inline src/lib/netlist/plib/pstring.cpp:86]: Spacing has gone weird on this one.

**Review by cuavas — CHANGES_REQUESTED** [review]
There’s one place where you’re accidentally casting a vector’s size type to its element type.  The rest of it looks fine.

File: src/lib/netlist/plib/pstring.cpp (hunk 2 above)
Comment (couriersud) [reply src/lib/netlist/plib/pstring.cpp]: Fixed.

**Pull request merged**